src/main/java/models/League.java

```diff
@@ -0,0 +1,46 @@
+package models;
+
+import java.util.LinkedList;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Queue;
+
+public class League {
+
+    private Queue<Team> initialTeams;
+    private HashMap<Integer, ArrayList<Team>> winners;
+
+    public League(Queue<Team> initialTeams){
+        this.initialTeams = initialTeams;
+        this.winners = new HashMap<>();
+    }
+
+    private void playRound(Queue<Team> teamsToPlay, int round){
+
+        Queue<Team> nextRoundTeams = new LinkedList<>();
+
+        if( winners.get(round) == null ){
+            winners.put(round, new ArrayList<>());
+        }
+
+        while( !teamsToPlay.isEmpty() ){
+            Match match = new Match(teamsToPlay.poll(), teamsToPlay.poll());
+            Team winner = match.getMatchWinner();
+
+            winners.get(round).add(winner);
+
+            nextRoundTeams.add(winner);
+        }
+
+        if( nextRoundTeams.size() > 1 ){
+            playRound(nextRoundTeams, round + 1);
```

Comment:
The function name `playRound` doesn't go very well with the recursiveness. I'd personally make a 2nd function that interatively calls `playRound` until there are not enough `nextRoundTeams`. You could also go for something like `playMatches` as name. Note that in the first case the function should probably return `nextRoundTeams`.